frontend/src/app/pagination.ts

```diff
@@ -13,14 +31,123 @@ export interface EventPaginationParams {
+  /** Stores the previously used parameters for reference. */
+  previousParams: Params | null = null;
+
+  /** Internal writeable signal that updates when a new page is loaded. */
+  private pageSignal: WritableSignal<Paginated<T, Params> | undefined> =
+    signal(undefined);
+  /** Signal that exposes the currently active pagination page to frontend services and components. */
+  public page = this.pageSignal.asReadonly();
+
+  /**
+   * Constructs a paginator object.
+   *
+   * @param api: The string of the API endpoint to be called when a new page is loaded.
+   */
+  constructor(
+    protected api: string,
+    protected http: HttpClient
+  ) {
+    this.api = api;
+  }
+
+  /**
+   * Loads a new pagination page based on the API endpoint provided in the constructor and provided
+   * pagination parameters.
+   *
+   * Usage:
+   * ```
+   * let paginator: Paginator<Organization, OrganizationPaginationParams> = new Paginator<>('api/organization');
+   * paginator.loadPage<>(params);
+   * paginator.page(); // Returns the loaded page.
+   * ```
+   *
+   * This method also supports a pagination operator function in the case that the API endpoint returns
+   * a model that is different than the provided type `T` for the paginator. This is to be most commonly
+   * used with converting `Json` repsonse models to the regular typed response models. To support this,
+   * the .loadPage method supports a optional generic type for the API response type.
+   *
+   * Usage:
+   * ```
+   * let paginator: Paginator<Event, EventPaginationParams> = new Paginator<>('/api/event');
+   * paginator.loadPage<EventJson>(params, eventJsonToEventOperatorFunction);
+   * paginator.page(); // Returns the loaded page, in type `Paginated<T>`
+   * ```
+   *
+   * @template APIType: (Optional) Response model from the API call, if it is different than `T`.
+   * @param paramStrings: Pagination parameters.
+   * @param operator: (Optional) Operator to convert data from `Paginated<APIType>` to `Paginated<T>`.
+   */
+  loadPage<APIType = T>(
+    paramStrings: Params,
+    operator?: PaginationOperatorFunction<APIType, T, Params> | null
+  ) {
+    // Stpres the previous pagination parameters used
+    this.previousParams = paramStrings;
+
+    // Determines the query for the URL based on the new paramateres.
+    let query = new URLSearchParams(paramStrings);
+    let route = this.api + '?' + query.toString();
+
+    // Determine if an operator function is necessary
+    if (operator) {
+      // If so, call the API, pipe it through the operator, and update the signal.
+      this.http.get<Paginated<APIType, Params>>(route).pipe(
+        operator,
+        tap((pageData) => this.pageSignal.set(pageData))
+      );
+    } else {
+      // Otherwise, just call the API and update the signal.
+      this.http
+        .get<Paginated<T, Params>>(route)
+        .pipe(tap((pageData) => this.pageSignal.set(pageData)));
+    }
```

Comment:
Do these not need to be returned so that they can be subscribed to?

Reply:
I do not believe so, since they are only used to update the internal signal! This follows the same convention we have used in admin services with `RxObject`, where the `.get()` methods just update the internal state rather than returning an observable.

Reply:
@KrisJordan now, if we instead wanted the services themselves to hold the signal for the data, then we could just return the observable and use `tap` in the service to update its signal - that would work too!